engine/common/follower/cache/cache.go

```diff
@@ -198,17 +249,13 @@ func (c *Cache) PruneUpToView(view uint64) {
 //   - requires pre-computed blockIDs in the same order as fullBlocks
 //
 // Any errors are symptoms of internal state corruption.
-func (c *Cache) unsafeAtomicAdd(blockIDs []flow.Identifier, fullBlocks []*flow.Block) (batchContext, error) {
+func (c *Cache) unsafeAtomicAdd(blockIDs []flow.Identifier, fullBlocks []*flow.Block) (bc batchContext) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-	bc := batchContext{}
 
 	// add blocks to underlying cache, check for equivocation and report if detected
 	for i, block := range fullBlocks {
```

Comment:
How do you feel about reordering caching and checking for parent & child of batch. It's possible that due to caching we will eject possible parent or child that could have been resolved with this batch. To it seems relatively safe to change the order.